src/main/kotlin/za/ac/sun/plume/drivers/OverflowDbDriver.kt

```diff
@@ -0,0 +1,255 @@
+
+    fun connect() {
+        graph = createEmptyGraph()
+    }
+
+    private fun createEmptyGraph() : Graph {
+        val odbConfig = if (dbfilename != "") {
+            Config.withDefaults().withStorageLocation(dbfilename)
+        } else {
+            Config.withDefaults()
+        }
+        return Graph.open(odbConfig,
+                io.shiftleft.codepropertygraph.generated.nodes.Factories.allAsJava(),
+                io.shiftleft.codepropertygraph.generated.edges.Factories.allAsJava())
+    }
+
+    override fun addVertex(v: PlumeVertex) {
+        val id = v.hashCode()
+        val node = convert(v)
+        val newNode = graph.addNode(id.toLong(), node.label())
+        node.properties().foreachEntry { key, value ->
+            newNode.setProperty(key, value)
+        }
+    }
+
+    private fun convert(v : PlumeVertex) : NewNode {
+        return when(v) {
+            is ArrayInitializerVertex -> arrayInitializer(v.order)
+            is BindingVertex -> binding(v.name, v.signature)
+            is BlockVertex -> block(v.code, v.columnNumber, v.lineNumber, v.order, v.typeFullName, v.argumentIndex)
+            is CallVertex -> call(v.code, v.name, v.columnNumber, v.lineNumber, v.order, v.methodFullName, v.argumentIndex, v.signature, v.dispatchType.name, v.dynamicTypeHintFullName, v.typeFullName)
+            is ControlStructureVertex -> controlStructure(v.code, v.columnNumber, v.lineNumber, v.order)
+            is FileVertex -> file(v.name, v.order)
+            is IdentifierVertex -> identifier(v.code, v.name, v.columnNumber, v.lineNumber, v.order, v.typeFullName, v.argumentIndex)
+            is JumpTargetVertex -> jumpTarget(v.code, v.name, v.columnNumber, v.lineNumber, v.order)
+            is LiteralVertex -> literal(v.code, v.columnNumber, v.lineNumber, v.order, v.typeFullName, v.argumentIndex)
+            is LocalVertex -> local(v.code, v.name, v.columnNumber, v.lineNumber, v.order, v.typeFullName)
+            is MetaDataVertex -> metaData(v.language, v.version)
+            is MethodVertex ->
+                method(v.code, v.name, v.fullName, v.signature, v.order, v.columnNumber, v.lineNumber)
+            is MethodParameterInVertex -> methodParameter(v.code, v.name, v.lineNumber, v.order, v.evaluationStrategy.name, v.typeFullName)
+            is MethodReturnVertex -> methodReturn(v.code, v.columnNumber, v.lineNumber, v.order, v.typeFullName, v.evaluationStrategy.name)
+            is NamespaceBlockVertex -> namespaceBlock(v.name, v.fullName, v.order)
+            is ReturnVertex -> returnNode(v.code, v.lineNumber, v.order, v.argumentIndex)
+            is TypeVertex -> NewType(v.name, v.fullName, v.typeDeclFullName)
+            is TypeArgumentVertex -> typeArgument(v.order)
+            is TypeDeclVertex -> typeDecl(v.name, v.fullName, v.order, v.typeDeclFullName)
+            is TypeParameterVertex -> typeParameter(v.name, v.order)
+            else -> {
+               println(v)
```

Comment:
We probably want to handle this more gracefully and remove the println.

Reply:
Maybe add a logger for now e.g.
```kotlin
import org.apache.logging.log4j.LogManager
...
private val logger = LogManager.getLogger(OverflowDbDriver::class.java)
```
Then on the else branch log it as:
```kotlin
else -> { logger.warn("Received unsupported vertex type."); null }
```

I would then change the return type of `convert` to `NewNode?`. After this, made `addVertex` null-safe by changing the code to:
```kotlin
override fun addVertex(v: PlumeVertex) {
        val id = v.hashCode()
        convert(v)?.let { node ->
           val newNode = graph.addNode(id.toLong(), node.label())
           node.properties().foreachEntry { key, value ->
               newNode.setProperty(key, value)
           }
        }
    }
```

Reply:
Implemented as suggested.